lib/vdsm/storage/storageServer.py

```diff
@@ -924,11 +924,22 @@ def createConnection(cls, conInfo):
 
 
 def connect(dom_type, con_defs):
+    if not con_defs:
+        log.warning("Connect: connection list is empty, dom_type=%s", dom_type)
+
     con_class, connections = _prepare_connections(dom_type, con_defs)
     return con_class.connect_all(connections)
 
 
 def disconnect(dom_type, con_defs):
+    if not con_defs:
+        log.warning(
+            "Disconnect: connection list is empty, dom_type=%s", dom_type)
+        # When connection list is empty, there's nothing to disconnect. Engine
+        # currently sends empty list when more than one iSCSI SD use same iSCSI
+        # target. Return empty result list.
+        return []
```

Comment:
Again this is not the place to fix. Assert that con_defs are not empty.

The fix should be in hsm - this is input validation issue. If engine sends
invalid input, we need to warn and return without doing anything.

Reply:
done